crates/re_log_types/src/path/entity_path_filter.rs

```diff
@@ -459,19 +484,58 @@ mod tests {
             "/world/car/driver",
             "/x/y/z",
         ];
-        let rules = rules.map(EntityPathRule::parse_forgiving);
+        let rules = rules.map(|rule| EntityPathRule::parse_forgiving(rule, &subst_env));
         check_total_order(&rules);
     }
 
     #[test]
     fn test_entity_path_filter() {
+        let subst_env = Default::default();
+
         let filter = EntityPathFilter::parse_forgiving(
             r#"
         + /world/**
         - /world/
         - /world/car/**
         + /world/car/driver
         "#,
+            &subst_env,
+        );
+
+        for (path, expected_effect) in [
+            ("/unworldly", None),
+            ("/world", Some(RuleEffect::Exclude)),
+            ("/world/house", Some(RuleEffect::Include)),
+            ("/world/car", Some(RuleEffect::Exclude)),
+            ("/world/car/hood", Some(RuleEffect::Exclude)),
+            ("/world/car/driver", Some(RuleEffect::Include)),
+            ("/world/car/driver/head", Some(RuleEffect::Exclude)),
+        ] {
+            assert_eq!(
+                filter.most_specific_match(&EntityPath::from(path)),
+                expected_effect,
+                "path: {path:?}",
+            );
+        }
+
+        assert_eq!(
+            EntityPathFilter::parse_forgiving("/**", &subst_env).formatted(),
+            "+ /**"
+        );
+    }
+
+    #[test]
+    fn test_entity_path_filter_subs() {
+        let subst_env = std::iter::once(("origin".to_owned(), "/world".to_owned())).collect();
```

Comment:
It may be worth testing that we can also substitute for a path that is longer than one (e.g. `$origin = /what/a/beautiful/world`)